src/main/java/com/iexec/sms/tee/session/base/SecretSessionBaseService.java

```diff
@@ -141,81 +125,98 @@ public Map<String, Object> getSecretsTokens(TeeSecretsSessionRequest request) th
             throws TeeSessionGenerationException {
+        SecretEnclaveBaseBuilder enclaveBase = SecretEnclaveBase.builder();
+        enclaveBase.name("pre-compute");
+        Map<String, Object> tokens = new HashMap<>();
         TaskDescription taskDescription = request.getTaskDescription();
         String taskId = taskDescription.getChainTaskId();
-        Map<String, Object> tokens = new HashMap<>();
-        String fingerprint = teeWorkflowConfig.getPreComputeFingerprint();
-        tokens.put(PRE_COMPUTE_MRENCLAVE, fingerprint);
+        enclaveBase.mrenclave(teeWorkflowConfig.getPreComputeFingerprint());
+        tokens.put(IEXEC_PRE_COMPUTE_OUT, IexecFileHelper.SLASH_IEXEC_IN);
+        // `IS_DATASET_REQUIRED` still meaningful?
         tokens.put(IS_DATASET_REQUIRED, taskDescription.containsDataset());
-        tokens.put(IEXEC_DATASET_KEY, EMPTY_YML_VALUE);
+
+        List<String> trustedEnv = new ArrayList<>();
         if (taskDescription.containsDataset()) {
             String datasetKey = web3SecretService
                     .getSecret(taskDescription.getDatasetAddress(), true)
                     .orElseThrow(() -> new TeeSessionGenerationException(
                             PRE_COMPUTE_GET_DATASET_SECRET_FAILED,
-                            "Empty dataset secret - taskId: " + taskId
-                    ))
+                            "Empty dataset secret - taskId: " + taskId))
                     .getTrimmedValue();
             tokens.put(IEXEC_DATASET_KEY, datasetKey);
+            trustedEnv.addAll(List.of(
+                    IexecEnvUtils.IEXEC_DATASET_URL,
+                    IexecEnvUtils.IEXEC_DATASET_FILENAME,
+                    IexecEnvUtils.IEXEC_DATASET_CHECKSUM));
         } else {
             log.info("No dataset key needed for this task [taskId:{}]", taskId);
         }
-        // extract <IEXEC_INPUT_FILE_URL_N, url>
-        // this map will be empty (not null) if no input file is found
-        Map<String, String> inputFileUrls = IexecEnvUtils.getAllIexecEnv(taskDescription)
+        trustedEnv.addAll(List.of(
+                IexecEnvUtils.IEXEC_TASK_ID,
+                IexecEnvUtils.IEXEC_INPUT_FILES_FOLDER,
+                IexecEnvUtils.IEXEC_INPUT_FILES_NUMBER));
+        Map<String, String> trustedEnvVars = IexecEnvUtils.getAllIexecEnv(taskDescription)
                 .entrySet()
                 .stream()
-                .filter(e -> e.getKey().contains(IexecEnvUtils.IEXEC_INPUT_FILE_URL_PREFIX))
+                .filter(e ->
+                // extract trusted en vars to include
+                trustedEnv.contains(e.getKey())
+                        // extract <IEXEC_INPUT_FILE_URL_N, url>
+                        || e.getKey().contains(IexecEnvUtils.IEXEC_INPUT_FILE_URL_PREFIX))
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        tokens.put(INPUT_FILE_URLS, inputFileUrls);
-        return tokens;
+        tokens.putAll(trustedEnvVars);
```

Comment:
It may be faster & better to not store the env vars in a temporary map? Something like:
```java
.forEach(e -> tokens.put(e.getKey(), e.getValue()))
```

Reply:
Thanks! Done here: https://github.com/iExecBlockchainComputing/iexec-sms/pull/112/commits/f698c65c0e71317eb2226b97ed1f532c68588bd4